src/00/z2ui5_cl_fw_utility.clas.abap

```diff
@@ -508,94 +470,143 @@ CLASS z2ui5_cl_fw_utility IMPLEMENTATION.
   ENDMETHOD.
 
 
-  METHOD url_param_create_url.
+  METHOD get_t_attri_by_obj.
 
-    LOOP AT t_params INTO DATA(ls_param).
-      result = result && ls_param-n && `=` && ls_param-v && `&`.
-    ENDLOOP.
-    result = shift_right( val = result
-                          sub = `&` ).
+    DATA(lo_obj_ref) = cl_abap_objectdescr=>describe_by_object_ref( val ).
+    result  = CAST cl_abap_classdescr( lo_obj_ref )->attributes.
+    DELETE result WHERE visibility <> cl_abap_classdescr=>public OR is_interface = abap_true.
 
   ENDMETHOD.
 
 
-  METHOD url_param_get.
+  METHOD get_t_attri_by_struc.
 
-    DATA(lt_params) = url_param_get_tab( url ).
-    DATA(lv_val) = get_trim_lower( val ).
-    result = VALUE #( lt_params[ n = lv_val ]-v OPTIONAL ).
+    FIELD-SYMBOLS <attribute> TYPE any.
 
-  ENDMETHOD.
+    DATA(lv_name) = `IO_APP->` && to_upper( iv_attri ).
+    ASSIGN (lv_name) TO <attribute>.
+    raise( when = xsdbool( sy-subrc <> 0 ) ).
 
+    DATA(lo_type) = cl_abap_structdescr=>describe_by_data( <attribute> ).
+    DATA(lo_struct) = CAST cl_abap_structdescr( lo_type ).
 
-  METHOD url_param_get_tab.
+    DATA(lv_attri) = iv_attri.
+    DATA(lv_length) = strlen( lv_attri ) - 2.
+    DATA(lv_attri_end) = lv_attri+lv_length.
 
-    DATA(lv_search) = replace( val  = i_val sub  = `%3D` with = '=' occ  = 0 ).
-    lv_search = shift_left( val = lv_search sub = `?` ).
-    lv_search = get_trim_lower( lv_search ).
+    IF lv_attri_end = `>*`.
+      lv_attri_end = `>`.
+      lv_length = lv_length.
+      lv_attri = lv_attri(lv_length) && lv_attri_end.
+    ELSE.
+      lv_attri_end = `-`.
+      lv_length = lv_length + 2.
+      lv_attri = lv_attri(lv_length) && lv_attri_end.
+    ENDIF.
```

Comment:
```suggestion
    IF lv_attri_end = `>*`.
      lv_attri_end = `>`.
      lv_length = lv_length.
    ELSE.
      lv_attri_end = `-`.
      lv_length = lv_length + 2.
    ENDIF.
    lv_attri = lv_attri(lv_length) && lv_attri_end.    
```

Reply:
thanks, just sow it now and added it.
working on a bigger change here, still need to review and fix the code a few more times.